### PR TITLE
rename `hearth-types` to `hearth-schema`

### PR DESCRIPTION
**Pull request by airidaceae**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -17,7 +17,7 @@ hearth-macros.path = "crates/hearth-macros"
 hearth-network.path = "crates/hearth-network"
 hearth-rend3.path = "crates/hearth-rend3"
 hearth-terminal.path = "crates/hearth-terminal"
-hearth-types.path = "crates/hearth-types"
+hearth-schema.path = "crates/hearth-schema"
```
Comment (marceline-cramer) [inline Cargo.toml:20]: Please move this up a line to maintain the alphabetical order.

File: README.md [hunk 2]
```diff
@@ -142,15 +142,15 @@ to be aware of:
     loading, the lump store, and runtime building. Because the `hearth-core`
     API is essential to most Hearth functionality, most crates in the Hearth
     codebase depend on it.
-- **hearth-types**: a schema crate that defines the guest-to-host message
+- **hearth-schema**: a schema crate that defines the guest-to-host message
```
Comment (marceline-cramer) [inline README.md:145]: Nice catch.

File: crates/hearth-client/Cargo.toml [hunk 3]
```diff
@@ -15,7 +15,7 @@ hearth-init = { workspace = true }
 hearth-network = { workspace = true }
 hearth-rend3 = { workspace = true }
 hearth-terminal = { workspace = true }
-hearth-types = { workspace = true }
+# hearth-schema = { workspace = true }
```
Comment (marceline-cramer) [inline crates/hearth-client/Cargo.toml:18]: Might as well remove this.

File: guest/rust/hearth-guest/Cargo.toml [hunk 4]
```diff
@@ -6,7 +6,7 @@ license = "AGPL-3.0-or-later"
 
 [dependencies]
 bitflags = { workspace = true }
-hearth-types = { workspace = true }
+hearth-schema = { workspace = true }
```
Comment (marceline-cramer) [inline guest/rust/hearth-guest/Cargo.toml:9]: Props for catching this. Have you tried building and running kindling locally?

**Review by marceline-cramer — CHANGES_REQUESTED** [review]
Good work. :) A couple of super minor details but overall great stuff.

File: guest/rust/hearth-guest/Cargo.toml (hunk 4 above)
Comment (airidaceae) [reply guest/rust/hearth-guest/Cargo.toml:9]: yes! it seems to work as intended.

File: crates/hearth-client/Cargo.toml (hunk 3 above)
Comment (airidaceae) [reply crates/hearth-client/Cargo.toml:18]: oops! i did not mean to commit the commented version. But yeah, ill remove it.

**Review by marceline-cramer — APPROVED** [review]
Great work.

**Pull request merged**